pkg/test/pods.go

```diff
@@ -96,6 +106,13 @@ func Pod(overrides ...PodOptions) *v1.Pod {
 	}
 }
 
+func sequentialRandomName() string {
+	sequentialPodNumberLock.Lock()
+	defer sequentialPodNumberLock.Unlock()
```

Comment:
Since https://github.com/uber-go/atomic is already a dependency, you could just use
`var sequentialPodNumber atomic.Int64`
	and then where you use it, call sequentialPodNumber.Inc() instead and there won't be a need for the mutex.  I like the static part being first, makes it easy to find your pod!

Reply:
I'm not sure why that is a dependency, we don't seem to be using it in the code base: https://github.com/aws/karpenter/search?q=%22go.uber.org%2Fatomic%22

Maybe it's a dependency of a dependency?

For something basic like this I rather use the standard Go library over a third party library because the cost of ownership is much less.

Reply:
Yeah, it's a dependency of a dependency.  `go mod graph` lists it, but I'm good either way.